modules/nixos/base.nix

```diff
@@ -80,9 +81,20 @@
         [
           ../common/user.nix
           ./systemd.nix
-          ({name, ...}: let
-            user = config.users.users.${name};
+          ({
+            config,
+            name,
+            ...
+          }: let
+            user = osConfig.users.users.${name};
           in {
+            assertions = [
+              {
+                assertion = config.enable -> user.enable;
+                message = "Enabled Hjem user '${name}' must also be configured and enabled in NixOS.";
+              }
+            ];
+
```

Comment:
I recall having this in one of the earliest prototypes of Hjem (or something similar, at least) but we ended up dropping the reliance on NixOS for some reason. Do you remember why?

Not opposed to the change, I just want to make sure we're not re-introducing something we already patched out for good reason.